### PR TITLE
CFT - example solutions, IAM_member binding update

### PR DESCRIPTION
**Pull request by ocsig**
Adding example solutions how to use wrapper classes in the context of CFT.

### CODE REVIEW

**Comment by ocsig** [comment]
@sjvanrossum This PR is two separate solution. The IAM member update is a simplification of the existing template.

File: community/cloud-foundation/example-solutions/wrapper-template/folders-wrapper.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+
+def generate_config(context):
+
+    # Using some global values from an external config file.
+    # Hardcoded for this example.
+
+    global_prefic = "acc "
```
Comment (sjvanrossum) [inline community/cloud-foundation/example-solutions/wrapper-template/folders-wrapper.py]:
```suggestion
    global_prefix = "acc "
```

File: community/cloud-foundation/example-solutions/wrapper-template/folders-wrapper.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    # Manipulate context.properties #
+
+    for folder in context.properties["folders"]:
+        folder["displayName"] = global_prefic + folder["displayName"]
```
Comment (sjvanrossum) [inline community/cloud-foundation/example-solutions/wrapper-template/folders-wrapper.py]:
```suggestion
        folder["displayName"] = global_prefix + folder["displayName"]
```

File: community/cloud-foundation/example-solutions/wrapper-template/generic-wrapper.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  name: generic-wrapper.py
+- path: ../templates/project/project.py
+  name: target-template.py
+
+resources:
+- name: cft_project
+  type: generic-wrapper.py
+  properties:
+    environment: dev
+    module: project
```
Comment (sjvanrossum) [inline community/cloud-foundation/example-solutions/wrapper-template/generic-wrapper.yaml]:
```suggestion
    module: project

```

File: community/cloud-foundation/templates/iam_member/iam_member.py [hunk 4]
```diff
@@ -18,61 +18,24 @@ def generate_config(context):
...
-                'resource': project_id,
-            }
-        },
-        {
-            # Set the IAM policy by patching the existing policy with the
-            # config contents.
-            'name': policy_add_name,
-            'action': 'gcp-types/cloudresourcemanager-v1:cloudresourcemanager.projects.setIamPolicy',  # pylint: disable=line-too-long
-            'properties':
+        for member in role['members']:
```
Comment (sjvanrossum) [inline community/cloud-foundation/templates/iam_member/iam_member.py]:
```suggestion
        for i, member in enumerate(role['members']):
```

File: community/cloud-foundation/templates/iam_member/iam_member.py [hunk 5]
```diff
@@ -18,61 +18,24 @@ def generate_config(context):
     """ Entry point for the deployment resources. """
 
     project_id = context.env['project']
-    policy_get_name = 'get-iam-policy-{}'.format(project_id)
-    policy_add_name = 'add-iam-policy-{}'.format(project_id)
-    policy_remove_name = 'remove-iam-policy-{}'.format(project_id)
 
-    policies_to_add = []
+    resources = []
+    i = 0
```
Comment (sjvanrossum) [inline community/cloud-foundation/templates/iam_member/iam_member.py]:
```suggestion
```

File: community/cloud-foundation/templates/iam_member/iam_member.py [hunk 6]
```diff
@@ -18,61 +18,24 @@ def generate_config(context):
...
-                    'gcpIamPolicyPatch':
-                        {
-                            # Remove the previously created roles.
-                            'remove': policies_to_add
-                        }
-                }
-        }
-    ]
+            )
+            i += 1
```
Comment (sjvanrossum) [inline community/cloud-foundation/templates/iam_member/iam_member.py]:
```suggestion
```

**Pull request merged**